File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        for c in s:
            if c == '(' or c == '{' or c == '[':
                stack.append(c)
            else:
                if c == ')':
                    if not stack or stack.pop()!='(':
                        return False
                elif c == '}':
                    if not stack or stack.pop()!='{':
                        return False
                elif c == ']':
                    if not stack or stack.pop()!='[':
                        return False
        if not stack:
            return True
        return False
        

File: 0020-valid-parentheses/0020-valid-parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        std::stack<char> stack;
        for(int i=0;i<s.length();i++){
            if(s[i]=='(' || s[i]=='{' || s[i]=='[')
                stack.push(s[i]);
            else{
                if(s[i]==')'){
                    if(stack.empty() || stack.top()!='(')
                        return false;
                    stack.pop();
                }else if(s[i]=='}'){
                    if(stack.empty() || stack.top()!='{')
                        return false;
                    stack.pop();
                }else if(s[i]==']'){
                    if(stack.empty() || stack.top()!='[')
                        return false;
                    stack.pop();
                } 
            }
        }
        return stack.empty();
    }
};